spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java

```diff
@@ -153,6 +153,8 @@ public void undeploy(String id) {
 				}
 			}
 			running.remove(id);
+		} else {
+			throw new IllegalStateException("There is nothing to un-deploy for deployment id " + id);
```

Comment:
There is no special assertion on the message. But, the exception message is changed from this commit: https://github.com/spring-cloud/spring-cloud-deployer-local/commit/34a5bb96d818b4277e677edb800b680fa20846c8 and similar update happened across deployer implementations.